core/src/main/java/org/jboss/galleon/runtime/PackageRuntime.java

```diff
@@ -164,10 +164,34 @@ static Builder builder(FeaturePackRuntimeBuilder fp, PackageSpec spec, Path dir,
     private final PackageSpec spec;
     private final Path layoutDir;
 
+    private final int status;
+    private final boolean isPassiveIncluded;
+    private final boolean passive;
+
     private PackageRuntime(Builder builder, FeaturePackRuntime fp) {
         this.fp = fp;
         this.spec = builder.spec;
         this.layoutDir = builder.dir;
+        passive = builder.type == PackageDependencySpec.PASSIVE;
+        isPassiveIncluded = passive
+                && builder.isPassiveWithSatisfiedDeps();
+        status = builder.status;
+    }
+
+    public boolean isPassive() {
+        return passive;
+    }
+
+    public boolean isRequired() {
+      return (status & INCLUDED) > 0;
```

Comment:
Given that we use required and optional to describe the nature of a dependency, I don't think we should re-use them to express something else. A good idea would be to ask native English speakers for an advice. Here are a few alternatives I could think of for an installed package that could safely be removed: extra, spare, frivolous.